packages/synapse-react-client/src/components/TimelinePlot/TimelinePhase.tsx

```diff
@@ -215,11 +219,54 @@ const TimelinePhase = ({
   const selectedRows = rowData?.filter(row => {
     return rowIds?.includes(row.rowId)
   })
+
   const hoverEventRowIds = hoverEvent?.points[0].customdata as unknown as (
     | number
     | undefined
   )[]
 
+  const types = selectedRows.map(row => row.values[schema.observationType])
+
+  const isValidJSONString = (str: string | null | undefined): str is string => {
+    if (typeof str !== 'string') return false
+    try {
+      JSON.parse(str)
+      return true
+    } catch {
+      return false
+    }
+  }
+
+  const flattenedData: string[] = Array.from(
+    new Set(
+      types
+        .filter(isValidJSONString)
+        .map(item => JSON.parse(item))
+        .flat(),
+    ),
+  )
+
+  const items = [
+    ...flattenedData.map(type => ({
+      text: type || '',
+      onClick: () => {
+        setSelectedObservationTypes(prev => {
+          if (prev.includes(type)) {
+            return prev.filter(t => t !== type)
+          } else {
+            return [...prev, type]
+          }
+        })
+      },
+    })),
+  ]
+
+  const filteredRows = rowData.filter(row =>
+    selectedObservationTypes.every(type =>
+      JSON.parse(row.values[schema.observationType] ?? '[]').includes(type),
+    ),
+  )
```

Comment:
Instead of filtering client-side, we should make this new UI control apply a facet filter and pass it to Synapse in the QueryBundleRequest.

To actually get the valid observationType, the [Observations MaterializedView](https://www.synapse.org/Synapse:syn51735464/tables/) should be updated to make `observationType` a faceted column. Once that's done, Synapse will give us the set of observationType values in the QueryResultBundle.facets object.